File: src/nl/hannahsten/texifyidea/file/FileNameMatcher.java
package nl.hannahsten.texifyidea.file;

/**
 * @author Hannah Schellekens
 */
@FunctionalInterface
public interface FileNameMatcher {

    /**
     * Checks if the given file name matches or not.
     *
     * @param fileName
     *         The name of the file to match.
     * @return {@code true} if the fileName matches, {@code false} when the fileName does not match.
     */
    boolean matchesName(String fileName);
}

File: src/nl/hannahsten/texifyidea/file/FileNameMatcher.kt
package nl.hannahsten.texifyidea.file

/**
 * @author Hannah Schellekens
 */
@FunctionalInterface
interface FileNameMatcher {

    /**
     * Checks if the given file name matches or not.
     *
     * @param fileName
     * The name of the file to match.
     * @return `true` if the fileName matches, `false` when the fileName does not match.
     */
    fun matchesName(fileName: String): Boolean
}